cidr-filtering-cni

```diff
@@ -0,0 +1,252 @@
+    local chain="$3"
+    local hook="$4"
+    local device="$5"
+    echo "nft add chain $type $name $chain { type filter hook $hook device $device priority 0; policy accept; }" | tee -a $logFile
+}
+
+create_base_chain() {
+    local type="$1"
+    local name="$2"
+    local chain="$3"
+    local hook="$4"
+    echo "nft add chain $type $name $chain { type filter hook $hook priority 0; policy accept; }" | tee -a $logFile
+}
+
+create_chain() {
+    local type="$1"
+    local name="$2"
+    local chain="$3"
+    echo "nft add chain $type $name $chain" | tee -a $logFile
+}
+
+nft_add_rule() {
+    local type="$1"
+    local table="$2"
+    local chain="$3"
+    set -- "${@:4}"
+    echo "nft add rule $type $table $chain $@" | tee -a $logFile
+}
+
+create_rules_for_filtering() {
+    local policy_id="$1"
+    local table_type="$2"
+    local direction="$3"
+    local match_iface="$4"
+    
+    local match_addr="saddr"
+    if [[ "$direction" == "egress" ]]; then
+        local match_addr="daddr"
+    fi
+    
+    _create_subnet_rule() {
+        local subnet=$1
+        if [[ "$(get_object "${subnet}" ".subnet.except")" != "null" ]]; then
+            for_json_array "${subnet}" ".subnet.except" _drop_ip
+        fi
+
+        if [[ "$(get_object "${subnet}" ".subnet.cidr")" == "null" ]]; then
+            exitWithError "cidr must be specified."
+        fi
+        _accept_cidr "${subnet}"
+    }
+
+    _drop_ip() {
+        local ip="$1"
+        ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-subnets ${match_iface} $CNI_IFNAME $(get_ip_version "${ip}") ${match_addr} ${ip} counter drop)
+    }
+
+    _accept_cidr() {
+        local subnet="$1"
+        local cidr=$(echo "${subnet}" | jq -r ".subnet.cidr")
+        ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-subnets ${match_iface} $CNI_IFNAME $(get_ip_version "${cidr}") ${match_addr} ${cidr} counter meta mark set mark or 0x20000)
+    }
+
+    _accept_port() {
+        local ports="$1"
+        local port=$(echo "$ports" | jq -r ".port")
+        local protocol=$(echo "$ports" | jq -r ".protocol")
+
+        if [[ "$port" == "null" || "$protocol" == "null" || "$port" == "" || "$protocol" == "" ]]; then
+            exitWithError "Port and protocol must be specified."
+        else
+            ip netns exec "$CNI_CONTAINERID" $(nft_add_rule ${table_type} ${NFT_TABLE} m-${policy_id}-${direction}-ports ${match_iface} $CNI_IFNAME ${protocol,,} dport ${port} counter meta mark set meta mark "|" 0x00010000)
+        fi
+    }
+
+    # handle ip block
+    if [[ $(echo "$stdin" | jq -r ".${direction}.subnets") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.subnets" _create_subnet_rule
+    fi
+
+    # handle ports
+    if [[ $(echo "$stdin" | jq -r ".${direction}.ports") != "null" ]]; then
+        for_json_array "$stdin" ".${direction}.ports" _accept_port
+    fi
```

Comment:
so, to make sure I understand: the ports would apply for **all** the IP blocks, right ?

Reply:
Yes, that is correct